src/library/scala/collection/BitSet.scala

```diff
@@ -140,6 +140,62 @@ trait BitSetOps[+C <: BitSet with BitSetOps[C]]
 
   override def isEmpty: Boolean = 0 until nwords forall (i => word(i) == 0)
 
+  override def max[B >: Int](implicit ord: Ordering[B]): Int =
+    if (Ordering.Int eq ord) {
+      var i = nwords - 1
+      var currentWord = 0L
+      while(i >= 0) {
+        currentWord = word(i)
+        if (currentWord != 0L) {
```

Comment:
Small while loops optimize better.  I'd move this out, actually.  No harm doing a comparison twice.  (You may want to save the var after all if you do, though.)